### PR TITLE
[hdf4] Fix build on Darwin, which blocked gdal, postgis and more

### PR DESCRIPTION
**Pull request by purcell**
The libtirpc library is not available on Darwin, which led to hdf4 failing to build and subsequently blocking dependent packages.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by purcell** [comment]
Pinging @knedlsepp, who added this dep to fix aarch64 a couple of days ago.

**Comment by knedlsepp** [comment]
@ofborg build hdf4

**Comment by knedlsepp** [comment]
> Pinging @knedlsepp, who added this dep to fix aarch64 a couple of days ago.

Sorry about that, didn't test for Darwin.

**Review by knedlsepp — CHANGES_REQUESTED** [review]
Code looks fine. Could you fix the commit message format to comply to the [guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)

```
(pkg-name | nixos/<module>): (from -> to | init at version | refactor | etc)
```

**Comment by purcell** [comment]
> Could you fix the commit message format to comply to the guidelines

Oops, yes, done.

**Comment by purcell** [comment]
With the commit message fixed and all the tests clear, I'll go ahead and merge, since the code looks good to you.

**Comment by purcell** [comment]
Oh, I can't merge it myself, doh. Must not have write access yet, only gained org membership recently.

**Comment by knedlsepp** [comment]
> Oh, I can't merge it myself, doh. Must not have write access yet, only gained org membership recently.

Same here, otherwise I would have merged.

**Comment by purcell** [comment]
Maybe the ever-present and helpful @marsam would be kind enough to merge this? :-)

**Review by marsam — APPROVED** [review]

**Comment by purcell** [comment]
Yay, thanks @marsam

**Pull request merged**
